lib/parse.js

```diff
@@ -120,45 +143,59 @@ function parsePlistXML (node) {
     // TODO: what should we do with text types? (CDATA sections)
 
   } else if (node.nodeName === 'key') {
-    if(isEmptyNode(node)) return null;
     res = '';
-    if(node.childNodes[0]) {
+    if(!isEmptyNode(node)) {
       res = node.childNodes[0].nodeValue;
     }
     return res;
   } else if (node.nodeName === 'string') {
     res = '';
-    if(isEmptyNode(node)) return null;
-    for (d=0; d < node.childNodes.length; d++) {
-      res += node.childNodes[d].nodeValue;
+    if(!isEmptyNode(node)) {
+      for (i=0; i < node.childNodes.length; i++) {
+        var type = node.childNodes[i].nodeType;
+        if (type === 3 || type === 4) {
+          res += node.childNodes[i].nodeValue;
+        }
+      }
     }
     return res;
 
   } else if (node.nodeName === 'integer') {
-    // parse as base 10 integer
+    invariant(
+      !isEmptyNode(node),
+      'Cannot parse "" as integer.'
+    );
     return parseInt(node.childNodes[0].nodeValue, 10);
 
   } else if (node.nodeName === 'real') {
+    invariant(
+      !isEmptyNode(node),
+      'Cannot parse "" as real.'
+    );
     res = '';
-    for (d=0; d < node.childNodes.length; d++) {
-      if (node.childNodes[d].nodeType === 3) {
-        res += node.childNodes[d].nodeValue;
+    for (i=0; i < node.childNodes.length; i++) {
+      if (node.childNodes[i].nodeType === 3) {
```

Comment:
I know you didn't write this code; it pre-dates your additions. But while you're in here, can you replace the magic number usage here too?